akka-projection-r2dbc/src/main/scala/akka/projection/r2dbc/internal/R2dbcOffsetStore.scala

```diff
@@ -656,8 +682,9 @@ private[projection] class R2dbcOffsetStore(
       }
     if (newInflight.size >= 10000) {
```

Comment:
I have also tested this in a real service where I logged the inFlight size and could see that it was increasing over time. After the fix it is 0 after each save offsets, as expected.